src/jdk.javadoc/share/classes/jdk/javadoc/internal/package-info.java

```diff
@@ -24,21 +24,142 @@
+ *          {@linkplain jdk.javadoc.internal.doclets.toolkit.AbstractDoclet abstract doclet},
+ *          which uses
+ *          {@linkplain jdk.javadoc.internal.doclets.toolkit.builders.AbstractBuilder builders}
+ *          to generate pages of abstract
+ *          {@linkplain jdk.javadoc.internal.doclets.toolkit.Content content}.
+ *
+ *          <p>The format-specific content for each page is provided by implementations
+ *          of various <em>writer</em> interfaces, created by a format-specific
+ *          {@linkplain jdk.javadoc.internal.doclets.toolkit.WriterFactory writer factory}.
+ *
+ *          <p>The {@link jdk.javadoc.internal.doclets.toolkit.BaseConfiguration} provides
+ *          configuration information that is relevant to all the generated pages.
+ *          Some of the information is provided by abstract methods which are implemented
+ *          in format-specific subtypes of the class.
+ *
+ *          <p>The toolkit layer also provides
+ *          {@linkplain jdk.javadoc.internal.doclets.toolkit.util utility classes}
+ *          used by this layer and by format-specific layers.
+ *
+ *          <p id="workarounds">Generally, it is intended that doclets should use the
+ *          {@link javax.lang.model Language Model API} to navigate the structure of a Java program,
+ *          without needing to access any internal details of the underlying <em>javac</em> implementation.
+ *          However, there are still some shortcomings of the Language Model API,
+ *          and so it is still necessary to provide limited access to some of those internal details.
+ *          Although not enforceable by the module system, by design the access to <em>javac</em>
+ *          internal details by doclets based on {@code AbstractDoclet} is restricted to the aptly-named
+ *          {@link jdk.javadoc.internal.doclets.toolkit.WorkArounds} class.
+ *
+ *        <dt>Other Doclets
+ *        <dd><p>
+ *          Doclets are obviously not required to use
+ *          {@link jdk.javadoc.internal.doclets.toolkit.AbstractDoclet} and other classes in
+ *          the toolkit layer. In times past, it was common to write doclets to analyze
+ *          code using the then-current API as an early version of a Java language model.
+ *          That old API has been replaced by the {@link javax.lang.model Language Model API},
+ *          and tools that wish to use that API to analyze Java programs have a choice of
+ *          how to invoke it, using the <em>javac</em> support for
+ *          {@linkplain javax.annotation.processing annotation processing},
+ *          or {@linkplain com.sun.source.util.Plugin plugins}, as well as doclets.
+ *          Which is best for any application will depend of the circumstances, but
+ *          if a tool does not need access to the documentation comments in a program,
+ *          it is possible that one of the other invocation mechanisms will be more convenient.
```

Comment:
Should the "Other Doclets" part above be added to the "Doclet Interface" section below? I think they deal with very related topics. The text above goes into some detail about when and how to implement doclets, which is the topic of the text below. Also, the list above mostly describes the actual components of the javadoc module, but "Other Doclets" is not really part of that.

Reply:
I'll take a look.

Reply:
I think it helps to keep them separate ... and the text is adjacent.

"Other Doclets" is somewhat more about why you might not want to write a doclet these days ... it's addressing the historical past where there were lots of doclets, a lot of which did not even look at doc comments ... i.e they were using the Doclet API as an early Language Model API.

Reply:
Fine with me. Your explanation makes sense, another bit of history learned.

Reply:
I'd forgotten, until now, there is a more detailed version of this text on the OpenJDK website.  I'll leave all the text here, but add an `@see`